### PR TITLE
Fix newly created bug [PLAT-951]

### PR DESCRIPTION
**Pull request by nc-firthh**
https://nextcapital.atlassian.net/browse/PLAT-951

This branch fixes a bug introduced by stories that do not have a sprint, status, or resolution change in their changelog.

### CODE REVIEW

File: src/scripts/components/jira/overall-progress/OverallProgressBody.jsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ const OverallProgressBody = (props) => (
           _.map(_.take(props.highlights, config.highlightLimit), (highlight) => (
             <OverallProgressHighlight
               key={ highlight.fields.id }
+              createdDate={ highlight.lastChange ? highlight.lastChange.created : highlight.fields.created }
```
Comment (nc-rheew) [inline src/scripts/components/jira/overall-progress/OverallProgressBody.jsx]: maybe do something like this
`createdDate={ _.get(highlight, 'lastChange.created') || highlight.fields.created }`

File: src/scripts/components/jira/overall-progress/OverallProgressHighlight.jsx [hunk 2]
```diff
@@ -8,17 +8,21 @@ import config from './config';
...
+  if (props.change) {
+    const newSprint = _.find(props.change.items, { fieldId : 'customfield_10113' });
+    const newStatus = _.find(props.change.items, { fieldId : 'status' });
+
+    const hasBeenAdded = newSprint && newSprint.to == props.currentSprintId;
+
+    if (newStatus) {
+      changeText = config.status.messages[props.statusCategory.key];
+    } else {
+      changeText = hasBeenAdded ?
```
Comment (nc-rheew) [inline src/scripts/components/jira/overall-progress/OverallProgressHighlight.jsx]: if `hasBeenAdded` is only being used here, I would just replace it with `newSprint && newSprint.to == props.currentSprintId`

File: src/scripts/components/jira/overall-progress/OverallProgressHighlight.jsx [hunk 3]
```diff
@@ -8,17 +8,21 @@ import config from './config';
...
-    changeText = hasBeenAdded ?
-      config.sprint.messages.addedText :
-      config.sprint.messages.removedText;
+  if (props.change) {
+    const newSprint = _.find(props.change.items, { fieldId : 'customfield_10113' });
+    const newStatus = _.find(props.change.items, { fieldId : 'status' });
+
+    const hasBeenAdded = newSprint && newSprint.to == props.currentSprintId;
+
+    if (newStatus) {
```
Comment (nc-rheew) [inline src/scripts/components/jira/overall-progress/OverallProgressHighlight.jsx]: If you're not planning on using `hasBeenAdded` anywhere else, I would just replace this with `if (_.find(props.change.items, { fieldId : 'status' }))`

File: src/scripts/slides/jira_slide.jsx [hunk 4]
```diff
@@ -251,7 +251,7 @@ const JiraSlide = React.createClass({
       );
     } else if (this.state.view === VIEWS.overall.key) {
       const highlights = _.sortBy(_.flatten(this.state.stories), (story) => {
-        return story.lastChange.created;
+        return story.lastChange ? story.lastChange.created : story.fields.created;
```
Comment (nc-rheew) [inline src/scripts/slides/jira_slide.jsx]: i think you can just do `(story) => _.get(story, 'lastChange.created') || story.fields.created`

File: src/scripts/slides/jira_slide.jsx [hunk 5]
```diff
@@ -264,7 +264,7 @@ const JiraSlide = React.createClass({
     }
 
     let lastChangeDate = _.max(
-      _.map(_.flatten(this.state.stories), (story) => story.lastChange.created)
+      _.map(_.flatten(this.state.stories), (story) => story.lastChange ? story.lastChange.created : story.fields.created)
```
Comment (nc-rheew) [inline src/scripts/slides/jira_slide.jsx]: same

Comment (nc-firthh) [inline src/scripts/slides/jira_slide.jsx]: TIL about `_.get()`

Can't believe I've gone this long without it.

**Pull request merged**